oat-sa/tao-release-notes
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417)

### Summary

Update the dependencies for supporting Node 18.

### Details

To properly support Node 18, the dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

It also removes most of the vulnerabilities 🎉 
For removing the last ones, the code must be refactored to use ESM and then we can upgrade to the latest versions of dependencies.
```
audited 720 packages in 2.496s

found 4 vulnerabilities (2 moderate, 2 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

### How to test
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`
- test the linter: `npm run lint`
- install globally `npm link`, then check the release note using composer files:
    - download https://raw.githubusercontent.com/oat-sa/tao-community/2023.03/composer.json as `composer.2023.03.json` (ex: `curl https://raw.githubusercontent.com/oat-sa/tao-community/2023.03/composer.json -o composer.2023.03.json`)
    - download https://raw.githubusercontent.com/oat-sa/tao-community/2023.04/composer.json as `composer.2023.04.json` (ex: `curl https://raw.githubusercontent.com/oat-sa/tao-community/2023.04/composer.json -o composer.2023.04.json`)
    - run `taoReleaseNotes composers --c1 composer.2023.03.json --c2 composer.2023.04.json -f md`



[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
